utils/kubernetes/kubeutil.py

```diff
@@ -140,33 +209,33 @@ def retrieve_metrics(self):
         """
         return retrieve_json(self.metrics_url)
 
-    def filter_pods_list(self, pods_list, host_ip):
+    def retrieve_kubelet_url(self, url, verbose=True, timeout=10):
```

Comment:
This is not an explicit name for what it does. This sounds like it's returning a url as a string.